README.md

```diff
@@ -1,26 +1,57 @@
-# Offscreen Effect Player module
+# [Offscreen Effect Player module (OEP-module)](https://docs.banuba.com/face-ar-sdk-v1/core/cpp_api_overview#offscreen-effect-player)
+
+[Description of the **Offscreen Effect Player** on the Banuba SDK website](https://docs.banuba.com/face-ar-sdk-v1/core/cpp_api_overview#offscreen-effect-player)
 
 ## Description
 
-Common submodule for some projects using **offscreen_effect_player**
-The module implements interaction with the effect player. The module consists of six main interfaces:
+Common submodule for some projects using [**Offscreen effect player**](./interfaces/offscreen_effect_player.hpp). The module implements interaction with the effect player.
+
+## Folders structure
+
+- [**docs**](./docs/) - illustrations used in this README.md
+- [**interfaces**](./interfaces/) - offscreen effect player interfaces
+- [**offscreen_effect_player**](./offscreen_effect_player/) - is a wrapper for effect_player. It allows you to use your own implementation for offscreen_render_target
+- [**offscreen_render_target**](./offscreen_render_target/) - is an implementation option for the offscreen_render_target interface. Allows to prepare gl framebuffers and textures for receiving a frame from gpu, receive bytes of the processed frame from the gpu and pass them to the cpu, as well as, if necessary, set the orientation for the received frame. This implementation uses GLFW to work with gl context
+- [**opengl**](./opengl/) - OpenGL utilities
+- [**third**](./third/) - third party libraries
+
+## Description of interfaces
+
+The module consists of six main interfaces.
+- [**effect_player**](./interfaces/effect_player.hpp) - implementation contains interaction with the effect player. Should be implemented on the application side.
+- [**image_processing_result**](./interfaces/image_processing_result.hpp) - the implementation contains the output image processor in the required format. Some formats are processed on GPU or CPU side.
+- [**offscreen_effect_player**](./interfaces/offscreen_effect_player.hpp) - the implementation connects all interfaces into a single whole. The main interface through which the entire pipeline for processing a stream of frames goes.
+- [**offscreen_render_target**](./interfaces/offscreen_render_target.hpp) - implementation creates and configures offscreen rendering. Uses the OpenGL renderer.
+- [**pixel_buffer**](./interfaces/pixel_buffer.hpp) - input and output pixel buffer (image). Contains an image in any of the supported formats: [RGB, RGBA, BGR, BGRA, ARGB, nv12, i420](./interfaces/image_format.hpp).
+- [**render_context**](./interfaces/render_context.hpp) - platform-specific rendering context. Should be implemented on the application side.
+
+## The scheme of interfaces interaction
+
+Direct arrows show how interfaces interact with each other. Dashed arrows show the sequence of data processing.
+![alt text](./docs/OEP-module-scheme.png)
```

Comment:
We should work together on the diagrams. In order to demonstrate interactions and relations usually are used separate diagrams.